src/Cache/Memory.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Jahuty\Cache;
+
+use Psr\SimpleCache\CacheInterface;
+use Jahuty\Exception\Cache as InvalidArgumentException;
+
+class Memory implements CacheInterface
```

Comment:
Since the min PHP version this library accepts is 7.3, we can use covariance here and add return types to these methods.

https://www.php.net/manual/en/language.oop5.variance.php

Reply:
👍 thanks! Done in the last commit.